cpp/src/datetime/datetime_ops.cu

```diff
@@ -75,26 +75,16 @@ struct extract_component_operator {
   }
 };
 
-CUDA_DEVICE_CALLABLE auto days_in_month(simt::std::chrono::month mon, bool is_leap_year) -> uint8_t
+// Number of days until month indexed by leap year and month (0-based index)
+static __device__ int16_t const days_until_month[2][13] = {
+  {0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334, 365},  // For non leap years
+  {0, 31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335, 366}   // For leap years
+};
+
+CUDA_DEVICE_CALLABLE uint8_t days_in_month(simt::std::chrono::month mon, bool is_leap_year)
 {
-  using namespace simt::std::chrono;
-  // The expression in switch has to be integral/enumerated type.
-  // The constexpr in case has to match the switch type
-  switch (unsigned{mon}) {
-    case unsigned{January}: return 31;
-    case unsigned{February}: return is_leap_year ? 29 : 28;
-    case unsigned{March}: return 31;
-    case unsigned{April}: return 30;
-    case unsigned{May}: return 31;
-    case unsigned{June}: return 30;
-    case unsigned{July}: return 31;
-    case unsigned{August}: return 31;
-    case unsigned{September}: return 30;
-    case unsigned{October}: return 31;
-    case unsigned{November}: return 30;
-    case unsigned{December}: return 31;
-    default: return 0;
-  }
+  return days_until_month[is_leap_year][unsigned{mon}] -
```

Comment:
I knew I remembered this table... I suggested it originally. This is a much better solution.